packages/wallet-protocols/src/protocols/create-and-direct-fund/protocol.ts

```diff
@@ -0,0 +1,120 @@
+import { Machine, MachineConfig, sendParent } from 'xstate';
+
+import { JsonRpcCreateChannelParams } from '../../json-rpc';
+import { MachineFactory } from '../../machine-utils';
+import { Store, success } from '../..';
+
+import { ethAllocationOutcome } from '../../calculations';
+
+import { AdvanceChannel, DirectFunding } from '..';
+
+const PROTOCOL = 'create-and-direct-fund';
+
+export enum Indices {
+  Left = 0,
+  Right = 0,
+}
+
+export type Init = JsonRpcCreateChannelParams & {
+  channelId: string;
+  challengeDuration: number;
+  index: Indices;
+};
+
+export type FirstStateConstructed = Init & { type: 'FIRST_STATE_CONSTRUCTED' };
+
+export const advanceChannelArgs = (i: 1 | 3) => ({ channelId }: Init): AdvanceChannel.Init => ({
+  channelId,
+  targetTurnNum: i,
+});
+
+const constructFirstState = {
+  invoke: {
+    id: 'constructFirstState',
+    src: 'constructFirstState',
+    data: (ctx: Init) => ctx,
+    onDone: 'preFundSetup',
+  },
+};
+
+const preFundSetup = {
+  invoke: {
+    id: 'preFundSetup',
+    src: 'advanceChannel',
+    data: advanceChannelArgs(1),
+    onDone: 'funding',
+  },
+  on: { CHANNEL_CLOSED: 'abort' },
+};
+
+// FIXME: Abort should not be success
+const abort = success;
```

Comment:
```suggestion
```

I think it's safe to just get rid of `abort` for now, as we do not yet have a plan for managing failures.